cwms-data-api/src/main/java/cwms/cda/data/dto/TimeSeries.java

```diff
@@ -248,6 +256,11 @@ public static class Record {
         @JsonProperty(value = "quality-code", index = 2)
         int qualityCode;
 
```

Comment:
Changed to a subclass. I added a custom deserializer to handle the different classes.

Reply:
Okay, so I said this, Adam said this, but after reading the code and our other discussions above timeseries does it make more sense to just make TimeSeries more generic?

e.g.

A builder where you manually add column names, index, and type, and and functions in the row builder to set such?
something like

```
withColumn(int index, String name, String description, Class<T> type) {
"logic"
}

... Record:

<T> setColumn(int index, T value, Class<T> type) {
  "logic"
}
```
Or something like that, it would prevent the need in TimeSeriesDaoImpl have two different loops doing almost 90% the same work. just a check for "I have this column requested, let's also add it."

Basically instead of hard coding the columns at all (okay, maybe time... is a time series) , the user of the given TimeSeries object (after set by builder) can define them at run-time.


Sorry, I know you did a lot here, that just came to me now looking through the current PR.